tests/foreman/ui/test_fact.py

```diff
@@ -24,34 +24,49 @@ def test_positive_upload_host_facts(
     module_sca_manifest_org,
     module_location,
     module_activation_key,
+    default_os,
 ):
-    """Verify Facts option is available on the Host UI and it is successfully updated in Satellite after registration
+    """Verify Facts option is available on the Host UI and it is successfully updated in Satellite after updated registration command.
 
     :id: f32093d2-4088-4025-9623-adb3141bd770
 
     :steps:
-        1. Register host to Satellite.
+        1. Register host with updated command to Satellite.
         2. Navigate to the Host page and click on kebab.
         3. Verify that the facts option is available and is updated on Satellite.
 
-    :expectedresults: Facts option is available on the Host UI and all the facts of the host are updated in Satellite.
+    :expectedresults:
+        1. The host is successfully registered using the label/name instead of the ID.
+        2. Facts option is available on the Host UI and all the facts of the host are updated in Satellite.
 
     :BZ: 2001552
 
+    :Verifies: SAT-28832
+
     :customerscenario: true
     """
+    hostgroup = module_target_sat.api.HostGroup(
+        organization=[module_sca_manifest_org], location=[module_location]
+    ).create()
     with module_target_sat.ui_session() as session:
         session.organization.select(module_sca_manifest_org.name)
         session.location.select(module_location.name)
         cmd = session.host.get_register_command(
             {
+                'general.operating_system': default_os.title,
+                'general.host_group': hostgroup.name,
                 'general.activation_keys': module_activation_key.name,
                 'general.insecure': True,
             }
         )
-        result = rhel_contenthost.execute(cmd)
+        result = cmd[cmd.find('hostgroup_id=') : cmd.find('&update_packages=')]
+        if ' ' in default_os.title:
+            default_os.title = default_os.title.replace(' ', '+')
+        new_data = f'hostgroup={hostgroup.name}&location={module_location.name}&operatingsystem={default_os.title}&organization={module_sca_manifest_org.name}'
+        updated_cmd = cmd.replace(result, new_data)
+        assert new_data in updated_cmd
+        result = rhel_contenthost.execute(updated_cmd)
```

Comment:
If we're switching the generated curl command manually for verifying this RFE, I'd suggest to create a new test for this in test_registration module, as it could impact the existing test and it looks unrelated to facts component, OR if I'm missing some context about this test please correct me

Reply:
+1 to create a new test for it.

Reply:
@Gauravtalreja1 @shweta83 
It was a small section, so I added it to the existing test instead of creating a new one. I don’t believe this change would impact the existing test since the PRT passed previously. Currently, it’s failing due to some internal issues, but otherwise, it should work. However, if you think I should create a new test for this, I’m happy to do so.

Reply:
I have created different method to cover automation, please take a look -